third_party/unix/ztypes_linux.go

```diff
@@ -0,0 +1,59 @@
+//go:build linux
+// +build linux
+
+/*
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// go run mksyscall.go -l32 -arm -tags linux,arm syscall_linux.go syscall_linux_arm.go
+// Code generated by the command above; see README.md. DO NOT EDIT.
+Modifies:
+- All the consts and types defined below are addition to the original code.
+*/
+
+package unix
+
+const (
+	IGMPMSG_NOCACHE  = 0x1
+	VIFF_USE_IFINDEX = 0x8
+	MRT_ADD_VIF      = 0xca
+	MRT_ADD_MFC      = 0xcc
+	MRT_DEL_MFC      = 0xcd
+	MRT_INIT         = 0xc8
+	MRT_TABLE        = 0xd1
+	MRT_FLUSH        = 0xd4
+	MAXVIFS          = 0x20
+)
+
+type Mfcctl struct {
+	Origin   [4]byte /* in_addr */
+	Mcastgrp [4]byte /* in_addr */
+	Parent   uint16
+	Ttls     [32]uint8
+	Pkt_cnt  uint32
+	Byte_cnt uint32
+	Wrong_if uint32
+	Expire   int32
+}
+type Vifctl struct {
+	Vifi        uint16
+	Flags       uint8
+	Threshold   uint8
+	Rate_limit  uint32
+	Lcl_ifindex int32
+	Rmt_addr    [4]byte /* in_addr */
+}
+
+const SizeofMfcctl = 0x3c
+const SizeofVifctl = 0x10
```

Comment:
The two variables are not used, are they needed?

Reply:
Updated.  I changed here to use these two consts
```
func SetsockoptViMfcctl(fd, level, opt int, mfcctl *Mfcctl) error {
	return setsockopt(fd, level, opt, unsafe.Pointer(mfcctl), SizeofMfcctl)
}

func SetsockoptVifctl(fd, level, opt int, vifctl *Vifctl) error {
	return setsockopt(fd, level, opt, unsafe.Pointer(vifctl), SizeofVifctl)
}
```
There are two ways of passing struct size in Golang sys repo https://github.com/golang/sys/blob/5a964db013201115fcba5c3d31ade965d0969335/unix/syscall_unix.go#L350 and https://github.com/golang/sys/blob/5a964db013201115fcba5c3d31ade965d0969335/unix/syscall_unix.go#L366. Honestly, I don't know the differences between them. I wrote a simple script to check if they are identical for `SizeofMfcctl` and `SizeofVifctl`.

```
package main

import(
        "fmt"
        "unsafe"
)

type Mfcctl struct {
        Origin   [4]byte /* in_addr */
        Mcastgrp [4]byte /* in_addr */
        Parent   uint16
        Ttls     [32]uint8
        Pkt_cnt  uint32
        Byte_cnt uint32
        Wrong_if uint32
        Expire   int32
}

type Vifctl struct {
        Vifi        uint16
        Flags       uint8
        Threshold   uint8
        Rate_limit  uint32
        Lcl_ifindex int32
        Rmt_addr    [4]byte /* in_addr */
}

const SizeofMfcctl = 0x3c
const SizeofVifctl = 0x10


func main(){
        var vifctl *Vifctl
        fmt.Printf("%t\n", int(unsafe.Sizeof(*vifctl)) == SizeofVifctl)
        var mfcctl *Mfcctl
        fmt.Printf("%t\n", int(unsafe.Sizeof(*mfcctl)) == SizeofMfcctl)
}
```
Run this script on arm, arm64, and amd64 platforms, all give positive results. Personally, I like to use consts SizeofMfcctl and SizeofVifctl because they are clearer.